apps/checkout-app/pages/channels/[channelId].tsx

```diff
@@ -25,7 +24,7 @@ const Channel = () => {
   const { appId, isAuthorized } = useAuthData();
   const [metafieldsQuery] = usePublicMetafieldsQuery({
     variables: {
-      id: appId || serverEnvVars.appId,
```

Comment:
I don't remember exactly, but probably server side rendering was not working properly without `serverEnvVars.appId`, as app-bridge (from which `appId` is taken) only works on client side. The problem could occur only in built app. Can you just make sure everything works as expected without it?

Reply:
It doesn't happen in built app.
We don't have any page in checkout-app that uses SSR. If we used one, we could instead of making this query:

```
query PrivateMetafields($id: ID!, $keys: [String!]) {
  app(id: $id) {
    id
    privateMetafields(keys: $keys)
  }
}
```

where you have to provide `id` of the app use this one:

```grpahql
query PrivateMetafieldsInfered($keys: [String!]) {
  app {
    id
    privateMetafields(keys: $keys)
  }
}
```

where you don't have to provide the id of the app, because it's inferred from `authToken` the app uses to communicate with Saleor (and this way we save the 'round-trip' of requests ;). This already what I use for API routes.

I've added a warning message if you actually try to use `appId` from `app-brdige` in SSR.
We shouldn't fetch the `appId` by default because if we'd do that then we would make a ton of requests made to Saleor when building the `checkout-app`, it would also drastically slow down the build time.

We're using the `appId` to only do the [Client-side data fetching](https://nextjs.org/docs/basic-features/data-fetching/client-side) which is fine, because at that time we have access to app-bridge data